File: content/problems/2807-insert-greatest-common-divisors-in-linked-list/solutions/single-pass-insert-between-original-nodes/Solution.kt
class Solution {
    private fun gcd(aInput:Int,bInput:Int):Int{
        var a=aInput;
        var b=bInput;
        while(b!=0){
            val next=a%b;
            a=b;
            b=next
        };
        return a
    };
    fun insertGreatestCommonDivisors(head:ListNode?):ListNode?{
        var node=head;
        while(node?.next!=null){
            val next=node.next!!;
            val inserted=ListNode(gcd(node.`val`,next.`val`));
            inserted.next=next;
            node.next=inserted;
            node=next
        };
        return head
    }
}

File: content/problems/2807-insert-greatest-common-divisors-in-linked-list/solutions/single-pass-insert-between-original-nodes/solution.cpp
class Solution {
    public:ListNode* insertGreatestCommonDivisors(ListNode* head){
        for(ListNode* node=head;node&&node->next;node=node->next->next)node->next=new ListNode(gcd(node->val,node->next->val),node->next);
        return head;
    }
};
